crates/sui-core/src/authority/authority_store.rs

```diff
@@ -550,24 +559,27 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
             std::iter::once((transaction_digest, certificate)),
         )?;
 
-        // Store the signed effects of the transaction
-        write_batch = write_batch.insert_batch(
-            &self.effects,
-            std::iter::once((transaction_digest, effects)),
-        )?;
-
         // Cleanup the lock of the shared objects.
         let write_batch =
             self.remove_shared_objects_locks(write_batch, transaction_digest, certificate)?;
 
         // Safe to unwrap since the "true" flag ensures we get a sequence value back.
-        self.batch_update_objects(
-            write_batch,
-            temporary_store,
-            *transaction_digest,
-            update_type,
-        )
-        .await
+        let res = self
+            .batch_update_objects(
+                write_batch,
+                temporary_store,
+                *transaction_digest,
+                update_type,
+            )
+            .await?;
+
```

Comment:
What happen if we crash here? Does it mean that we processed the transaction but that the effects are not stored? Can this lead to re-execution upon recover?

Reply:
To be precise, I mean we crash at line 575 (after calling update `batch_update_objects` )

Reply:
this will cause re-execution in that we will call execute_transaction_to_effects again.

But it doesn't resequence the tx - the lock service will reject the proposed sequence number and return the one that was committed earlier.